vllm/v1/engine/core.py

```diff
@@ -272,6 +277,33 @@ def step_with_batch_queue(self) -> Optional[EngineCoreOutputs]:
 
         return engine_core_outputs
 
+    def log_gpu_load_time(self, vllm_config: VllmConfig,
+                          elapsed: float) -> None:
+        """Calculate and log the total GPU model loading time.
+
+        Args:
+            vllm_config: The vLLM configuration.
+            elapsed: The elapsed time for engine initialization.
+        """
+        driver_worker = getattr(self.model_executor, "driver_worker", None)
+        model_runner = getattr(driver_worker, "model_runner", None)
+
+        if model_runner:
+            model_gpu_load_time = getattr(model_runner, "model_load_time", 0.0)
+            cuda_graph_capture_time = getattr(model_runner,
+                                              "cuda_graph_capture_time", 0.0)
```

Comment:
I really don't like all of these `getattr()s` - it's incredibly brittle since they'll just silently stop working and no-one will notice the metric is wrong.

Reply:
If we want these components to record these timings for consumption at the `EngineCore` level, we should add a well-defined interface for them - e.g. `self.model_executor.get_startup_stats()` or something